### ISSUE TITLE
[Question]: 使用paddle分布式训练文本分类模型，遇到的问题

### ISSUE THREAD

**Issue by fuqiang-benz** [PaddlePaddle/PaddleNLP#5035]
### 请提出你的问题

我是使用develop分支下文本分类代码：https://github.com/PaddlePaddle/PaddleNLP/tree/develop/applications/text_classification/multi_class
环境：
paddle = 2.4.1
paddlenlp = 2.5.0
python = 3.7

使用多块cpu训练命令进行训练：    python -m paddle.distributed.launch --nproc_per_node=6--backend='gloo' train.py \
    --do_train \
    --do_eval \
    --do_export \
    --model_name_or_path ernie-3.0-tiny-medium-v2-zh \
    --output_dir checkpoint \
    --device gpu \
    --num_train_epochs 100 \
    --early_stopping True \
    --early_stopping_patience 5 \
    --learning_rate 3e-5 \
    --max_length 128 \
    --per_device_eval_batch_size 32 \
    --per_device_train_batch_size 32 \
    --metric_for_best_model accuracy \
    --load_best_model_at_end \
    --logging_steps 5 \
    --evaluation_strategy epoch \
    --save_strategy epoch \
    --save_total_limit 1

内存也没有跑满

训练到百分之25的时候报错：
![1677655811103](https://user-images.githubusercontent.com/64388539/222072817-dc72a133-98df-47c3-913f-ccf4bf1ab805.png)

以下是训练的数据集：
[data.txt](https://github.com/PaddlePaddle/PaddleNLP/files/10858254/data.txt)
[dev.txt](https://github.com/PaddlePaddle/PaddleNLP/files/10858257/dev.txt)
[label.txt](https://github.com/PaddlePaddle/PaddleNLP/files/10858259/label.txt)
[train.txt](https://github.com/PaddlePaddle/PaddleNLP/files/10858261/train.txt)

麻烦大佬帮忙看一下，谢谢

**Comment by fuqiang-benz** [comment]
我用的是cpu训练，如果你们复现的话 ，上边device参数写错了，应该改成cpu。

**Comment by ZHUI** [comment]
好的，您有GPU吗？GPU上会有类似问题吗？看起来是CPU下通信的问题

**Comment by ZHUI** [comment]
cc: @ForFishes 亮哥有什么思路吗？

**Comment by fuqiang-benz** [comment]
> 好的，您有GPU吗？GPU上会有类似问题吗？看起来是CPU下通信的问题

单位没有GPU，我起了6个进程，感觉训练速度还没有一块cpu快，是不是进程起多了进程之间需要通信所以耗时就长了呢，进程少起两个会不会好一点？

**Comment by fuqiang-benz** [comment]
> cc: @ForFishes 亮哥有什么思路吗？

您好， 有什么好的解决办法吗

**Comment by lugimzzz** [comment]
这块可以考虑用aistudio上免费的gpu资源进行训练

**Comment by fuqiang-benz** [comment]
> 这块可以考虑用aistudio上免费的gpu资源进行训练

您好，合作的甲方目前只有cpu资源，有什么好的办法加快训练速度吗？

**Comment by lugimzzz** [comment]
训练这块cpu上加速有限，可以考虑选择比较小的模型比如ernie-3.0-tiny-pico-v2-zh，ernie-3.0-tiny-nano-v2-zh。建议还是训练的时候使用gpu资源，然后模型推理的时候可以用模型裁剪和int8量化策略加速在cpu上的推理速度

**Comment by fuqiang-benz** [comment]
> 训练这块cpu上加速有限，可以考虑选择比较小的模型比如ernie-3.0-tiny-pico-v2-zh，ernie-3.0-tiny-nano-v2-zh。建议还是训练的时候使用gpu资源，然后模型推理的时候可以用模型裁剪和int8量化策略加速在cpu上的推理速度

我记得paddle官方不是给了几个多块cpu训练的命令吗？python -m paddle.distributed.launch --nproc_per_node=6 --backend=gloo 还有 python -m paddle.distributed.launch --server_num=2 --trainer_num=2 为什么都不能用啊

**Comment by fuqiang-benz** [comment]
您好，我使用多块cpu训练一段时间报错，应该是进程之间通信的问题，有什么好的解决办法吗？执行的是上边多进程的命令
